module-2/backend/week_two.md

```diff
@@ -5,33 +5,71 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 8. Define foreign key, primary key, and schema.
+  foreign key is used to relate a database item to another database item. primary key is what a database item is primarily associated by
+  and the schema is the set of tables to describe each attribute of your database items.
 9. Describe the relationship between a foreign key on one table and a primary key on another table.
+  You might assign a foreign key to a table to associate or relate another table through its primary key.
+  like a Cars table might have a foreign key of tire_ids that would assign different primary key of a Tire object to that car.
 10. What are the parts of an HTTP response?
+  status line and status code?
 11. Describe some techniques to make our Sinatra code more DRY. Give an example of when you would use these techniques.
+  Breaking ActiveRecord requests you repeat into methods in your models. 
+  Also breaking your database into more Tables for items you may be repeating alot.
 
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
+  .group - takes a table and will group them into sub categories you can call on later
+  .pluck - allows you to pluck or grab a value by specifying the key, from a key , value pair set of data
+  .where - easiest search method, has lots of arguments you can pass it to get what you want.
+  .not - allows you to disclude items in a search
+  .order - allows you to order a returned list of items be a specified attribute.
 2. Name your three favorite ActiveRecord rake tasks and describe what they do.
+  rake db:create_migration - allows you to make/change a table in your database
+  rake db:migrate - grabs your migration tables and updates your schema
+  rake db:rollback - allows you to go back to the previous schema/database setup you had
 4. What can you expect from a group as you begin working together? As you continue working together?
+  I think, I would always expect communication. With communication you can always work through issues and stay on task.
 5. What two columns does `t.timestamps null: false` create in our database?
+  created_on and updated_on
 6. What cURL flag can you use to send a `POST` request?
+  -d
 7. What case does JSON (and JavaScript) use for multi-word variables?
+  camelCase?
 8. What case does Ruby use for multi-word variables?
+  snake_case
 9. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
+  schools have many teachers; teachers belong to a school
 10. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
+  schools: id, name, (has many :teachers); teachers: id, name, (belongs to :school)
 11. Give an example of when you might want to store information besides ids on a join table.
+  Im not sure
```

Comment:
Something like `grades` on an `enrollments` table. So a student has a grade from a teacher for each enrollment.